anesthetic/plot.py

```diff
@@ -787,7 +788,8 @@ def hist_plot_2d(ax, data_x, data_y, *args, **kwargs):
             pdf[pdf < cmin] = np.ma.masked
         if cmax is not None:
             pdf[pdf > cmax] = np.ma.masked
-        image = ax.pcolormesh(x, y, pdf.T, cmap=cmap, vmin=0, vmax=pdf.max(),
+        image = ax.pcolormesh(x, y, pdf.T, cmap=cmap,
+                              vmin=vmin, vmax=pdf.max(),
```

Comment:
Do we actually need the `vmax=pdf.max()` here? Could we remove it such that if someone wanted to fix it for some reason this would be possible?

If we do need it, do we also need it for `ax.hist2d` further up?

Reply:
Would it make sense to normalize the histogram like in the 1d case and set `vmax=1` by default? That would make the maximum consistent across a triangle plot (keeping in mind that in most cases there won't be a colorbar to go alongside the plot).

@Stefan-Heimersheim, @williamjameshandley: Thoughts?

Reply:
> `vmax=pdf.max()`

This seems to be the [default in `pcolormesh`](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.pcolormesh.html)
> **vmin, vmax**: float, default: None
> The colorbar range. If None, suitable min/max values are automatically chosen by the Normalize instance (defaults to the respective min/max values of C in case of the default linear scaling). It is deprecated to use vmin/vmax when norm is given.

So I think we can just leave the default if it doesn't do anything.


> Would it make sense to normalize the histogram like in the 1d case and set vmax=1 by default?

Doesn't this interfere with the y-axis in double-triangle plots where the y axis is actually relevant for the `upper` triangle and not necessarily 0 to 1?

Reply:
> the maximum consistent across a triangle plot (keeping in mind that in most cases there won't be a colorbar to go alongside the plot).

Oh I misunderstood the `vmax=1` idea. I see the idea, but I think it might be difficult if some PDFs are much narrower than others. Say you have N(sigma=0.1) and N(sigma=1) [not too uncommon], then the 2nd PDF would peak at most at 10% of the height of the plot and be very hard to read.

Thanks Lukas for making the PR!

Reply:
Thinking about it, I don't know anymore why I suggested that. Normalizing to maximum and setting `vmax=1` is no different to setting `vmax=pdf.max()` in all plots, so no need for that. 

But I do think that `vmax=pdf.max()` is the default anyhow, so we don't need to code it in there and by not putting it in explicitly we allow for `vmax` to be overwritten if desired, right? I've removed it in [babcf52](https://github.com/williamjameshandley/anesthetic/pull/157/commits/babcf5247c628e1c5103144941bb60fe22504b94).

Reply:
@Stefan-Heimersheim, could you confirm this achieves what you want, and if so approve the review so @lukashergt can squash and merge